docs/source/app-dev/services.rst

```diff
@@ -0,0 +1,159 @@
+
+Overview
+********
+
+The API is structured as two separate data streams:
+
+-  A stream of **commands** TO the ledger that allow an application to cause state changes.
+-  A stream of **events** FROM the ledger that indicate all state changes taking place on the ledger.
+
+Commands are the only way an application can cause the state of the ledger to change, and events are the only mechanism to read those changes.
+
+For an application, the most important consequence of these architectural decisions and implementation is that the ledger API is asynchronous. This means:
+
+-  The outcome of commands is only known some time after they are submitted.
+-  The application must deal with successful and erroneous command completions separately from command submission.
+-  Ledger state changes are indicated by events received asynchronously from the command submission that cause them.
+
+The application must handle these issues, and is a major determinant of application architecture. Understanding the consequences of the API characteristics is important for a successful application design.
+
+For more help understanding these issues so you can build correct, performant and maintainable applications, read the :doc:`application architecture guide </app-dev/app-arch>`.
+
+Submitting commands to the ledger
+*********************************
+
+Command submission service
+==========================
+
+Use the **command submission service** to submit a command to the ledger. Commands either create a new contract instance, or exercise a choice on an existing contract.
+
+A call to the command submission service will return as soon as the ledger server has parsed the command, and has either accepted or rejected it. This does not mean the command has been executed, only that the server has looked at the command and decided that it's format is acceptable, or has rejected it for syntactical or content reasons.
+
+The on-ledger effect of the command execution will be reported via an event delivered by the `transaction service <#transaction-service>`__, described below. The completion status of the command is reported via the `command completion service <#command-completion-service>`__. Your application should receive completions, correlate them with command submission, and handle errors and failed commands. 
+
+Commands can be labeled with two application-specific IDs, a :ref:`commandId <com.digitalasset.ledger.api.v1.Commands.command_id>`. and a :ref:`workflowId <com.digitalasset.ledger.api.v1.Commands.workflow_id>`, and both are returned in completion events. The ``commandId`` is returned to the submitting application only, and is generally used to implement this correlation between commands and completions. The ``workflowId`` is also returned (via a transaction event) to all applications receiving transactions resulting from a command. This can be used to track commands submitted by other applications.
```

Comment:
```suggestion
Commands can be labeled with two application-specific IDs, a :ref:`commandId <com.digitalasset.ledger.api.v1.Commands.command_id>`. and a :ref:`workflowId <com.digitalasset.ledger.api.v1.Commands.workflow_id>`, and both are returned in completion events. The ``commandId`` is returned to the submitting application only, and is generally used to implement this correlation between commands and completions. The ``workflowId`` is returned as part of the resulting transaction to all applications receiving it. This can be used to track workflows between parties, consisting of several transactions.
```